linera-client/src/client_context.rs

```diff
@@ -404,47 +404,45 @@ where
         chain_id: ChainId,
         message_id: MessageId,
         owner: AccountOwner,
-        validators: Option<Vec<(ValidatorPublicKey, String)>>,
     ) -> Result<(), Error>
     where
         S: Storage + Clone + Send + Sync + 'static,
     {
         let node_provider = self.make_node_provider();
-        let client = self.client.clone_with(
-            node_provider.clone(),
-            "Temporary client for fetching the parent chain",
-            vec![message_id.chain_id, chain_id],
-            false,
-        );
+        self.client.track_chain(chain_id);
 
         // Take the latest committee we know of.
         let admin_chain_id = self.wallet.genesis_admin_chain();
-        let query = ChainInfoQuery::new(admin_chain_id).with_committees();
-        let nodes: Vec<_> = if let Some(validators) = validators {
-            node_provider
-                .make_nodes_from_list(validators)?
-                .map(|(public_key, node)| RemoteNode { public_key, node })
-                .collect()
-        } else {
-            let info = client.local_node().handle_chain_info_query(query).await?;
-            let committee = info
-                .latest_committee()
-                .ok_or(error::Inner::ChainInfoResponseMissingCommittee)?;
-            node_provider
-                .make_nodes(committee)?
-                .map(|(public_key, node)| RemoteNode { public_key, node })
-                .collect()
-        };
+        let admin_client = self.make_chain_client(admin_chain_id)?;
+        let info = *admin_client.synchronize_from_validators().await?;
+        let committee = info
+            .epoch
+            .and_then(|epoch| {
+                info.requested_committees
+                    .as_ref()
+                    .and_then(|committees| committees.get(&epoch))
+            })
+            .ok_or(error::Inner::ChainInfoResponseMissingCommittee)?;
+        let nodes: Vec<_> = node_provider
+            .make_nodes(committee)?
+            .map(|(public_key, node)| RemoteNode { public_key, node })
+            .collect();
 
         // Download the parent chain.
         let target_height = message_id.height.try_add_one()?;
-        client
-            .download_certificates(&nodes, message_id.chain_id, target_height)
+        self.client
+            .download_certificates(
+                &nodes,
+                message_id.chain_id,
+                target_height,
+                committee.clone(),
+            )
```

Comment:
Can we just use `synchronize_from_validators` now, so we don't have to bother with the committee here? (Not sure if that's a bigger change or just works out of the box.)

…and use this new algorithm in `synchronize_chain_state`, too?